libs/tracing/unix/Tracing.ml

```diff
@@ -51,24 +51,81 @@ module Otel = Opentelemetry
  * TODO we'll probably need instructions for some system of tags?
  *)
 
+(*****************************************************************************)
+(* Types *)
+(*****************************************************************************)
+
+type analysis_flags = {
+  secrets_validators : bool;
+  historical_scan : bool;
+  allow_all_origins : bool;
+  deep_intra_file : bool;
+  deep_inter_file : bool;
+}
+
+type top_level_data = { version : string; analysis_flags : analysis_flags }
+
 (*****************************************************************************)
 (* Constants *)
 (*****************************************************************************)
 
 let default_endpoint = "https://telemetry.dev2.semgrep.dev"
 let endpoint_env_var = "SEMGREP_OTEL_ENDPOINT"
 
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+let oss_analysis () =
+  {
+    secrets_validators = false;
+    historical_scan = false;
+    allow_all_origins = false;
+    deep_intra_file = false;
+    deep_inter_file = false;
+  }
+
+(* Set the descriptor for allowed origins. This is not simply
+   a boolean because we will likely include new origins in the
+   future *)
+let allowed_origins allow_all_origins =
+  if allow_all_origins then "all_origins" else "pro_rules_only"
+
+(* Poor man's Git repo detection. Running git repo detection again
+   seems wasteful, but checking two env vars is pretty cheap.
+
+   TODO the more we port of semgrep scan and semgrep ci, the more
+   of this information will already be in OCaml *)
+let repo_name () =
+  match Sys.getenv_opt "SEMGREP_REPO_DISPLAY_NAME" with
+  | Some name -> name
+  | None -> (
+      match Sys.getenv_opt "SEMGREP_REPO_NAME" with
+      | Some name -> name
+      | None -> "<local run>")
+
+(* In case we don't have a repo name, report the base folder where
+   semgrep was run. We report only the base name to avoid leaking
+   user information they may not have expected us to include. *)
+let current_working_folder () = Filename.basename (Sys.getcwd ())
+
 (*****************************************************************************)
 (* Wrapping functions Trace gives us to instrument the code *)
 (*****************************************************************************)
 
+let enter_span = Trace_core.enter_span
+let exit_span = Trace_core.exit_span
 let with_span = Trace_core.with_span
 let add_data_to_span = Trace_core.add_data_to_span
 
+let add_data_to_opt_span sp data =
+  match sp with
+  | None -> ()
+  | Some sp -> Trace_core.add_data_to_span sp data
```

Comment:
```suggestion
let add_data_to_opt_span sp data =
  Option.iter (fun sp -> Trace_core.add_data_to_span sp data) sp
```

Reply:
Seems backwards this _requires_ the span to be optional. Would rather just do `Option.iter` at call sites of `add_data_to_span`.

Reply:
I would rather just have this than a special type for that. If you don't want to do it at the call sites this is fine I guess.